app.ts

```diff
@@ -0,0 +1,286 @@
+		if (row !== null) {
+			row.innerHTML += rowCols;
+		}
+	}
+}
+
+//// Fetch Requests
+// Response Error Handling
+function handleErrors(response: Response) {
+	if (!response.ok) {
+		throw Error(response.statusText);
+	}
+	return response;
+}
+
+// Heading Row (Getting the columns data)
+function getHeadings(): Promise<void> {
+	return fetch("http://localhost:2050/columns", {
+		method: "GET",
+		headers: { "Content-Type": "application/json" },
+	})
+		.then(handleErrors)
+		.then((response: Response) => response.json())
+		.then((headingData: string | string[]) => {
+			for (let heading of headingData) {
+				createHeadingColumn(heading);
+			}
+		})
+		.catch((error: Error) => {
+			console.log(error);
+		});
+}
+
+// Table Content (Getting the table's data)
+function getTable(): Promise<void> {
+	const content: HTMLElement | null = document.getElementById("content");
+	let toParameter = getParameters(fromParameter);
+	const pageStats: HTMLElement | null = document.getElementById("pageStats");
+
+	// Clears Table
+	if (content !== null) {
+		content.innerHTML = "";
+	}
+
+	// Displays The Current Results Being Shown
+	let currentStats = `Showing results from ${fromParameter} to ${toParameter} out of ${recordCount} results.`;
+	if (pageStats !== null) {
+		pageStats.innerHTML = currentStats;
+	}
+
+	return fetch(`http://localhost:2050/records?from=${fromParameter}&to=${toParameter}`, {
+		method: "GET",
+		headers: { "Content-Type": "application/json" },
+	})
+		.then(handleErrors)
+		.then((res: Response) => res.json())
+		.then((contentData: string | string[]) => {
+			for (let x of contentData) {
+				appendTableContent(x);
+			}
+		})
+		.catch((error: Error) => {
+			console.log(error);
+		});
+}
+
+// Gets Total Of All Records
+function getRecordCount(): Promise<void> {
+	return fetch("http://localhost:2050/recordCount", {
+		method: "GET",
+		headers: { "Content-Type": "application/json" },
+	})
+		.then(handleErrors)
+		.then((res: Response) => res.json())
+		.then((data: number) => {
+			recordCount = data;
+		})
+		.catch((error: Error) => {
+			console.log(error);
+		});
+}
+
+//// Debounce
+function debounce(fn: () => void, delay: number) {
+	let timer: number
+	return function () {
+		clearTimeout(timer);
+		timer = setTimeout(() => {
+			fn();
+		}, delay);
+	};
+}
+
+//// Sizing And Resizing
+function resizing() {
+	let end = fromParameter + getNoOfRows();
+	let toParameter: number;
+	let maxRecordsID = recordCount - 1;
+
+	if (end > maxRecordsID) {
+		toParameter = maxRecordsID;
+		fromParameter = toParameter - getNoOfRows();
+	}
+	getTable();
+}
+
+//// Navigation
+// Next
+class Next {
+	nextButton: HTMLElement | null;
+	nextTimer: number = 0
+
+	constructor() {
+		this.nextButton = document.getElementById("next");
+
+		const nextDebounce = (fn: () => void, delay: number) => {
+			return () => {
+				clearTimeout(this.nextTimer);
+				this.nextTimer = setTimeout(() => {
+					fn();
+				}, delay);
+			};
```

Comment:
I think you should rather change this to not return a function, that way you don't have to add that weird `()` on line 189.